lib/libcxx/include/__config

```diff
@@ -148,7 +148,7 @@
 #    endif
 // Feature macros for disabling pre ABI v1 features. All of these options
 // are deprecated.
-#    if defined(__FreeBSD__)
+#    if defined(__FreeBSD__) || defined(__DragonFly__)
```

Comment:
I'm fine with carrying this patch for now with the understanding that it will get wiped out when we update to LLVM 16. The DragonFlyBSD folks need to get their changes upstreamed into LLVM for a proper solution.